kernel/src/sched.rs

```diff
@@ -599,6 +619,11 @@ impl Kernel {
                 }
             }
         }
+        let time_executed_us = u32::min(
+            timeslice_us.unwrap_or(0) - systick.get_value(),
+            timeslice_us.unwrap_or(0),
+        ); //min is to protect from wrapping systick value
```

Comment:
Doesn't this calculate the time _not_ used? Or does systick count down? Or am I just not reading this right?

Reply:
The ARM systick counts down (#1985 renames `get_value()` to `get_time_remaining_us()`, which makes it clearer)